ports/espressif/supervisor/port.c

```diff
@@ -87,10 +89,25 @@
 
 #ifdef CONFIG_SPIRAM
 #include "esp32/spiram.h"
+#ifdef CONFIG_IDF_TARGET_ESP32
+#include "esp32/himem.h"
+#else
+#define esp_himem_reserved_area_size() (0)
+#endif
+
+static size_t spiram_size_usable_for_malloc(void) {
+    /* SPIRAM chip may be larger than the size we can map into address space */
+    size_t s = MIN(esp_spiram_get_size(), SOC_EXTRAM_DATA_SIZE);
+    return s - esp_himem_reserved_area_size();
+}
 #endif
 
 // Heap sizes for when there is no external RAM for CircuitPython to use
 // exclusively.
+#ifdef CONFIG_IDF_TARGET_ESP32
+// TODO: Determine better: 520kB of internal RAM; similar to 512kB for ESP32-S3.
+#define HEAP_SIZE (48 * 1024)
```

Comment:
Want to use 176 like the S3? Not 48 like the S2.

Reply:
allocating 176kB fails, which is why I changed it.